src/accountImporter.ts

```diff
@@ -258,65 +250,65 @@ var _validateRequiredParameters = function (options) {
     default:
       throw new FirebaseError("Unsupported hash algorithm " + clc.bold(options.hashAlgo));
   }
-};
+}
 
-var _validateProviderUserInfo = function (providerUserInfo) {
+function validateProviderUserInfo(providerUserInfo: any): any {
   if (!_.includes(ALLOWED_PROVIDER_IDS, providerUserInfo.providerId)) {
     return {
       error: JSON.stringify(providerUserInfo, null, 2) + " has unsupported providerId",
     };
   }
-  var keydiff = _.difference(_.keys(providerUserInfo), ALLOWED_PROVIDER_USER_INFO_KEYS);
+  const keydiff = _.difference(_.keys(providerUserInfo), ALLOWED_PROVIDER_USER_INFO_KEYS);
   if (keydiff.length) {
     return {
       error:
         JSON.stringify(providerUserInfo, null, 2) + " has unsupported keys: " + keydiff.join(","),
     };
   }
   return {};
-};
+}
 
-var validateUserJson = function (userJson) {
-  var keydiff = _.difference(_.keys(userJson), ALLOWED_JSON_KEYS);
+export function validateUserJson(userJson: any): any {
```

Comment:
Same here - this can return `{ error?: string }`

Reply:
wow, yeah, it can.